pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1071,11 +1083,30 @@ impl Timeline {
                 .into_iter()
                 .collect();
             rels.sort_unstable();
+            let mut relsize_keys_to_collect = KeySpaceRandomAccum::new();
             for rel in rels {
                 let relsize_key = rel_size_to_key(rel);
-                let mut buf = self.get(relsize_key, lsn, ctx).await?;
+                relsize_keys_to_collect.add_key(relsize_key);
+            }
+            // Skip the vectored-read max key check by using `get_vectored_impl`.
+            let io_concurrency = IoConcurrency::spawn_from_conf(
+                self.conf,
+                self.gate
+                    .enter()
+                    .map_err(|_| CollectKeySpaceError::Cancelled)?,
+            );
```

Comment:
I think `IoConcurrency::sequential()` mode is more appropriate here, for now.
This `io_concurrency` that you're creating here (= tokio task you'd be spawning here if concurrent IO is enabled) is too short-lived.